main.c

```diff
@@ -293,11 +355,19 @@ int main(int argc, char *argv[]) {
 		}
 	}
 
+	char *output_filename;
+	char tmp[64];
 	if (optind >= argc) {
-		fprintf(stderr, "%s", usage);
-		return EXIT_FAILURE;
+		default_filename(tmp, sizeof(tmp), output_filetype);
+		output_filename = tmp;
+	} else {
+		output_filename = argv[optind];
 	}
-	char *output_filename = argv[optind];
+
+	char *output_filepath;
+	char tmpd[256];
```

Comment:
No need for two variables here, `char output_filepath[PATH_MAX]` is enough. (`PATH_MAX` is the system-defined maximum path size)